packages/eas-cli/src/metadata/apple/config/writer.ts

```diff
@@ -0,0 +1,107 @@
+  constructor(public readonly schema: Partial<AppleMetadata> = {}) {}
+
+  /** Get the schema result to write it to the config file */
+  toSchema(): { configVersion: number; apple: Partial<AppleMetadata> } {
+    return {
+      configVersion: 0,
+      apple: this.schema,
+    };
+  }
+
+  setAgeRating(attributes: AttributesOf<AgeRatingDeclaration>): void {
+    this.schema.advisory = attributes;
+  }
+
+  setInfoLocale(attributes: AttributesOf<AppInfoLocalization>): void {
+    this.schema.info = this.schema.info ?? {};
+    const existing = this.schema.info[attributes.locale] ?? {};
+
+    this.schema.info[attributes.locale] = {
+      ...existing,
+      title: attributes.name ?? 'no name provided',
+      subtitle: optional(attributes.subtitle),
+      privacyPolicyUrl: optional(attributes.privacyPolicyUrl),
+      privacyPolicyText: optional(attributes.privacyPolicyText),
+      privacyChoicesUrl: optional(attributes.privacyChoicesUrl),
+    };
+  }
+
+  setCategories({ primaryCategory, secondaryCategory }: AttributesOf<AppInfo>): void {
+    this.schema.categories = [];
+
+    // TODO: see why these types are conflicting
+    if (primaryCategory) {
+      this.schema.categories.push(primaryCategory.id as any);
+      if (secondaryCategory) {
+        this.schema.categories.push(secondaryCategory.id as any);
+      }
+    }
+  }
+
+  setVersion(
+    attributes: Omit<AttributesOf<AppStoreVersion>, 'releaseType' | 'earliestReleaseDate'>
+  ): void {
+    this.schema.copyright = optional(attributes.copyright);
+  }
+
+  setVersionRelease(
+    attributes: Pick<AttributesOf<AppStoreVersion>, 'releaseType' | 'earliestReleaseDate'>
+  ): void {
+    if (attributes.releaseType === ReleaseType.SCHEDULED) {
+      this.schema.release = {
+        autoReleaseDate: optional(attributes.earliestReleaseDate),
+      };
+    }
+
+    if (attributes.releaseType === ReleaseType.AFTER_APPROVAL) {
+      this.schema.release = {
+        automaticRelease: true,
+      };
+    }
+
+    if (attributes.releaseType === ReleaseType.MANUAL) {
+      this.schema.release = {
+        automaticRelease: false,
+      };
+    }
+  }
+
+  setVersionLocale(attributes: AttributesOf<AppStoreVersionLocalization>): void {
+    this.schema.info = this.schema.info ?? {};
+    const existing = this.schema.info[attributes.locale] ?? {};
+
+    this.schema.info[attributes.locale] = {
+      ...existing,
+      description: optional(attributes.description),
+      keywords: optional(attributes.keywords)
+        ?.split(',')
+        .map(keyword => keyword.trim()),
+      releaseNotes: optional(attributes.whatsNew),
+      marketingUrl: optional(attributes.marketingUrl),
+      promoText: optional(attributes.promotionalText),
+      supportUrl: optional(attributes.supportUrl),
+    };
+  }
+}
+
+/** Helper function to convert `T | null` to `T | undefined`, required for the entity properties */
+function optional<T>(value: T | null): T | undefined {
```

Comment:
name does not tell you much about what it is doing, also you are not saving any space here, because
`optional(x)` is not much shorter than `x ?? undefined`

Reply:
It's not shorter, I agree with that but makes more sense when reading big chunks of code in the writer class. "optional attribute X"

Reply:
using wrapper function might be cleaner, but in this optional is not a good name, in most languages function optional would return abstraction that provide interface e.g. `hasValue(): bool`, `getValue(): T` where T is not nullable

in this case, your function is replacing nulls with undefined

Reply:
> in this case, your function is replacing nulls with undefined

Yes, that's what I meant to make clear here. I also tried to explain why in the comment below, but basically we have this:

- For apple utils, we need to use `null` values for non-values.
- For the `store.config.json` we need to omit writing these values so users won't have configs with all available properties written out (we'll default to sensible values when users don't write it)
- Some attributes can actually be `null`, so we have to do the `null -> undefined` conversion per attribute.

What would be your suggestion to name this? I think optional makes sense in this case, but I'm open to improvements :)

Reply:
> What would be your suggestion to name this

that's the point, I think doing that inline is cleaner because it's hard to name